### DISCUSSION TITLE
Controlling subsequence comparisons

### DISCUSSION THREAD

**Discussion by gerold-csendes-epam** [TDAmeritrade/stumpy#437] [Q&A] [answered]
Hello stumpy folks,
let's say I have a task for finding motifs where m=168 because we assume weekly (hour-level) pattern. For a downstream task, it may be useful to find motifs that match "week-by-week". So, instead of comparing all subsequences, we just compare specific ones e.g. Monday-Sunday to Monday-Sunday.
Is this possible in stumpy? I know this is a really specific task and after reading the docs I don't think it can be done. Still, I would be happy if someone could verify this.
Thanks a lot! :)

**Comment by seanlaw** [comment, marked as answer]
> Hello stumpy folks,
> let's say I have a task for finding motifs where m=168 because we assume weekly (hour-level) pattern. For a downstream task, it may be useful to find motifs that match "week-by-week". So, instead of comparing all subsequences, we just compare specific ones e.g. Monday-Sunday to Monday-Sunday.
> Is this possible in stumpy? I know this is a really specific task and after reading the docs I don't think it can be done. Still, I would be happy if someone could verify this.
> Thanks a lot! :)

@gerold-csendes-epam Before I can provide a response, I'd like to understand your question a little bit better using a more concrete example. Let's say we have 3 weeks worth of hourly data (i.e., `3 x 168 = 504` data points in total) stored in the `data` array. So, for the first week (i.e., `data[ : 168]`), you only want to compare (i.e., compute the distance between ) `data[ : 168]` to `data[168:336]` and then to `data[336:]`. Then, you take the `min` distance. Next, you repeat the same thing for `data[168:336]` and you compare this subsequence with `data[ : 168]` and `data[336:]`. Finally, you repeat the same thing for `data[336:]` and you compare this subsequence with `data[ : 168]` and `data[168:336]`. So, in total, there are only 6 distances being computed and no more. Is that correct? Otherwise, please clarify

**Reply by gerold-csendes-epam** [reply]
That is an accurate description, yes! Thanks @seanlaw

**Reply by seanlaw** [reply]
> Is this possible in stumpy? I know this is a really specific task and after reading the docs I don't think it can be done. Still, I would be happy if someone could verify this.

You are correct. This cannot be done with STUMPY and you are better off writing a nested `for-loop` to handle this since there is probably little performance gain to be had.

**Reply by gerold-csendes-epam** [reply]
great, that is good to know. Thanks a lot!

**Reply by seanlaw** [reply]
@gerold-csendes-epam Just an FYI that the new version was released today!
